lib/bot.js

```diff
@@ -183,7 +193,19 @@ class Bot {
       }
 
       const withMentions = text.replace(/@[^\s]+\b/g, (match) => {
-        const user = this.discord.users.find('username', match.substring(1));
+        const search = match.substring(1);
+        const guild = discordChannel.guild;
+        let user = guild.members.find('nickname', search);
+        if (!user) {
+          user = this.discord.users.find('username', search);
+          if (user) {
+            const serverUser = guild.members.find('id', user.id);
+            if (serverUser.nickname && serverUser.nickname !== search) {
+              // User has a guild nickname which is different from the search query
```

Comment:
Wouldn't `guild.members.find('id').nickname` just be null/undefined for users without nicknames? If so you'd just be comparing `search` to that anyway, which would never be true since search is the match from your regex - right?